src/project/mod.rs

```diff
@@ -290,6 +291,23 @@ impl Project {
             })
     }
 
+    /// Return the grouped environments, which are all solve-groups and the environments that need to be solved.
+    pub fn grouped_environments(&self) -> Vec<GroupedEnvironment> {
```

Comment:
These might not actually be unique. E.g. the return value might contain the same solvegroup or environment multiple times.

Reply:
Made it unique